File: Leetcode/690.cpp
/*
// Definition for Employee.
class Employee {
public:
    int id;
    int importance;
    vector<int> subordinates;
};
*/

constexpr int MAXN = 2005;

class Solution {
public:
    Solution() {
        ios_base::sync_with_stdio(false);
        cin.tie(nullptr);
        cout.tie(nullptr);
    }
    int id2importance[MAXN];
    vector<int> adj[MAXN];
    int sumOfImportance = 0;
    void dfs(int u) {
        sumOfImportance += id2importance[u];
        for (const auto &i : adj[u]) {
            dfs(i);
        }
    }
    int getImportance(vector<Employee*> employees, int id) {
        for (const auto &e : employees) {
            id2importance[e->id] = e->importance;
            for (const auto &i : e->subordinates) {
                adj[e->id].emplace_back(i);
            }
        }
        dfs(id);
        return sumOfImportance;
    }
};

File: Leetcode/690/690.go
/**
 * Definition for Employee.
 * type Employee struct {
 *     Id int
 *     Importance int
 *     Subordinates []int
 * }
 */

func dfs(id2employee *map[int]*Employee, id int, result *int) {
	temp := (*id2employee)[id]
	*result += temp.Importance
	for i := 0; i < len(temp.Subordinates); i++ {
		dfs(id2employee, temp.Subordinates[i], result)
	}
}

func getImportance(employees []*Employee, id int) int {
	id2employee := make(map[int]*Employee)
	for i := 0; i < len(employees); i++ {
		id2employee[employees[i].Id] = employees[i]
	}
	val, ok := id2employee[id]
	if !ok {
		return 0
	}
	result := val.Importance
	for i := 0; i < len(val.Subordinates); i++ {
		dfs(&id2employee, val.Subordinates[i], &result)
	}
	return result
}